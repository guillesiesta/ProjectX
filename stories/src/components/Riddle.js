import React, {Component} from 'react';

export default class Riddle extends Component {

  constructor(props){
    super(props);
    this.state={
      acertijo:'por defecto',
      pista1:'por defecto',
      pista2:'por defecto',
      pista3:'por defecto',
      estado:'por defecto',
    };
  }

componentDidMount(){
  fetch('http://localhost:5000/todo_por_titulo', {
        method: 'POST', // or 'PUT'
        body: JSON.stringify(this.props.titulo), // data can be `string` or {object}!
        headers:{
          'Content-Type': 'application/json'
        }})
    .then(response => response.json())
    .then(data => {
      console.log(data[0].short_storie) // Prints result from `response.json()` in getRequest
      console.log(data[0].pista1)
      console.log(data[0].pista2)
      console.log(data[0].pista3)
      console.log(data[0].estado)
      this.setState({acertijo:data[0].short_storie});
      this.setState({pista1:data[0].pista1});
      this.setState({pista2:data[0].pista2});
      this.setState({pista3:data[0].pista3});
      this.setState({estado:data[0].estado});


    })
    .catch(error => console.error(error))
}

/*cargaAcertijoEntero(){
  return(
    <p>{this.state.acertijo}</p>
    <ul class="list-group">
      <li class="list-group-item">{this.state.pista1}</li>
      <li class="list-group-item">{this.state.pista2}</li>
      <li class="list-group-item">{this.state.pista3}</li>
    </ul>
  )
}*/
  render(){
    return(
      <form>
        <div className="box-header with-border">
          <div className="row">
            <strong>{this.props.titulo}</strong>
            <p></p>
            <p className="text-success"><strong>{this.state.estado}%</strong></p>
            <p>{this.state.acertijo}</p>
            <ul className="list-group">
              <li className="list-group-item">{this.state.pista1}</li>
              <li className="list-group-item">{this.state.pista2}</li>
              <li className="list-group-item">{this.state.pista3}</li>
            </ul>
            <button type="submit" className="btn btn-default">Ver Comentarios</button>
          </div>
        </div>
      </form>
    );
  }

}
