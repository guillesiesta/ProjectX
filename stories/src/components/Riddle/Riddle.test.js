import React, {Component} from 'react';
import { shallow } from 'enzyme';

import Riddle from './Riddle';

import "isomorphic-fetch"

describe('Riddle', ()=> {
  test('Rendering the simple component', ()=>{

    const wrapper = shallow(
      <Riddle />
    );

    expect(wrapper).toMatchSnapshot();
  })
})
