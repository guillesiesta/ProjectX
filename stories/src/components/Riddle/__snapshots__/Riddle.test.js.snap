// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Riddle Rendering the simple component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Riddle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="box-header with-border"
      >
        <div
          className="row"
        >
          <strong />
          <p />
          <p
            className="text-success"
          >
            <strong>
              por defecto
              %
            </strong>
          </p>
          <p>
            por defecto
          </p>
          <ul
            className="list-group"
          >
            <li
              className="list-group-item"
            >
              por defecto
            </li>
            <li
              className="list-group-item"
            >
              por defecto
            </li>
            <li
              className="list-group-item"
            >
              por defecto
            </li>
          </ul>
          <textarea
            className="form-control"
            onChange={[Function]}
            rows="5"
            value=""
          />
          <p />
          <button
            className="btn btn-default"
            type="submit"
          >
            Enviar Solucion
          </button>
          <p />
          <br />
          <br />
          <br />
          <div>
            <div />
          </div>
          <p>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Ver Soluciones propuestas
            </button>
          </p>
        </div>
      </div>,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <strong />
          <p />
          <p
            className="text-success"
          >
            <strong>
              por defecto
              %
            </strong>
          </p>
          <p>
            por defecto
          </p>
          <ul
            className="list-group"
          >
            <li
              className="list-group-item"
            >
              por defecto
            </li>
            <li
              className="list-group-item"
            >
              por defecto
            </li>
            <li
              className="list-group-item"
            >
              por defecto
            </li>
          </ul>
          <textarea
            className="form-control"
            onChange={[Function]}
            rows="5"
            value=""
          />
          <p />
          <button
            className="btn btn-default"
            type="submit"
          >
            Enviar Solucion
          </button>
          <p />
          <br />
          <br />
          <br />
          <div>
            <div />
          </div>
          <p>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Ver Soluciones propuestas
            </button>
          </p>
        </div>,
        "className": "box-header with-border",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong />,
            <p />,
            <p
              className="text-success"
            >
              <strong>
                por defecto
                %
              </strong>
            </p>,
            <p>
              por defecto
            </p>,
            <ul
              className="list-group"
            >
              <li
                className="list-group-item"
              >
                por defecto
              </li>
              <li
                className="list-group-item"
              >
                por defecto
              </li>
              <li
                className="list-group-item"
              >
                por defecto
              </li>
            </ul>,
            <textarea
              className="form-control"
              onChange={[Function]}
              rows="5"
              value=""
            />,
            <p />,
            <button
              className="btn btn-default"
              type="submit"
            >
              Enviar Solucion
            </button>,
            <p />,
            <br />,
            <br />,
            <br />,
            <div>
              <div />
            </div>,
            <p>
              <button
                className="btn btn-default"
                onClick={[Function]}
                type="button"
              >
                Ver Soluciones propuestas
              </button>
            </p>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                por defecto
                %
              </strong>,
              "className": "text-success",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "por defecto",
                  "%",
                ],
              },
              "ref": null,
              "rendered": Array [
                "por defecto",
                "%",
              ],
              "type": "strong",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "por defecto",
            },
            "ref": null,
            "rendered": "por defecto",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="list-group-item"
                >
                  por defecto
                </li>,
                <li
                  className="list-group-item"
                >
                  por defecto
                </li>,
                <li
                  className="list-group-item"
                >
                  por defecto
                </li>,
              ],
              "className": "list-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "por defecto",
                  "className": "list-group-item",
                },
                "ref": null,
                "rendered": "por defecto",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "por defecto",
                  "className": "list-group-item",
                },
                "ref": null,
                "rendered": "por defecto",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "por defecto",
                  "className": "list-group-item",
                },
                "ref": null,
                "rendered": "por defecto",
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "onChange": [Function],
              "rows": "5",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enviar Solucion",
              "className": "btn btn-default",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Enviar Solucion",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-default"
                onClick={[Function]}
                type="button"
              >
                Ver Soluciones propuestas
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ver Soluciones propuestas",
                "className": "btn btn-default",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Ver Soluciones propuestas",
              "type": "button",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="box-header with-border"
        >
          <div
            className="row"
          >
            <strong />
            <p />
            <p
              className="text-success"
            >
              <strong>
                por defecto
                %
              </strong>
            </p>
            <p>
              por defecto
            </p>
            <ul
              className="list-group"
            >
              <li
                className="list-group-item"
              >
                por defecto
              </li>
              <li
                className="list-group-item"
              >
                por defecto
              </li>
              <li
                className="list-group-item"
              >
                por defecto
              </li>
            </ul>
            <textarea
              className="form-control"
              onChange={[Function]}
              rows="5"
              value=""
            />
            <p />
            <button
              className="btn btn-default"
              type="submit"
            >
              Enviar Solucion
            </button>
            <p />
            <br />
            <br />
            <br />
            <div>
              <div />
            </div>
            <p>
              <button
                className="btn btn-default"
                onClick={[Function]}
                type="button"
              >
                Ver Soluciones propuestas
              </button>
            </p>
          </div>
        </div>,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <strong />
            <p />
            <p
              className="text-success"
            >
              <strong>
                por defecto
                %
              </strong>
            </p>
            <p>
              por defecto
            </p>
            <ul
              className="list-group"
            >
              <li
                className="list-group-item"
              >
                por defecto
              </li>
              <li
                className="list-group-item"
              >
                por defecto
              </li>
              <li
                className="list-group-item"
              >
                por defecto
              </li>
            </ul>
            <textarea
              className="form-control"
              onChange={[Function]}
              rows="5"
              value=""
            />
            <p />
            <button
              className="btn btn-default"
              type="submit"
            >
              Enviar Solucion
            </button>
            <p />
            <br />
            <br />
            <br />
            <div>
              <div />
            </div>
            <p>
              <button
                className="btn btn-default"
                onClick={[Function]}
                type="button"
              >
                Ver Soluciones propuestas
              </button>
            </p>
          </div>,
          "className": "box-header with-border",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong />,
              <p />,
              <p
                className="text-success"
              >
                <strong>
                  por defecto
                  %
                </strong>
              </p>,
              <p>
                por defecto
              </p>,
              <ul
                className="list-group"
              >
                <li
                  className="list-group-item"
                >
                  por defecto
                </li>
                <li
                  className="list-group-item"
                >
                  por defecto
                </li>
                <li
                  className="list-group-item"
                >
                  por defecto
                </li>
              </ul>,
              <textarea
                className="form-control"
                onChange={[Function]}
                rows="5"
                value=""
              />,
              <p />,
              <button
                className="btn btn-default"
                type="submit"
              >
                Enviar Solucion
              </button>,
              <p />,
              <br />,
              <br />,
              <br />,
              <div>
                <div />
              </div>,
              <p>
                <button
                  className="btn btn-default"
                  onClick={[Function]}
                  type="button"
                >
                  Ver Soluciones propuestas
                </button>
              </p>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  por defecto
                  %
                </strong>,
                "className": "text-success",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "por defecto",
                    "%",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "por defecto",
                  "%",
                ],
                "type": "strong",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "por defecto",
              },
              "ref": null,
              "rendered": "por defecto",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="list-group-item"
                  >
                    por defecto
                  </li>,
                  <li
                    className="list-group-item"
                  >
                    por defecto
                  </li>,
                  <li
                    className="list-group-item"
                  >
                    por defecto
                  </li>,
                ],
                "className": "list-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "por defecto",
                    "className": "list-group-item",
                  },
                  "ref": null,
                  "rendered": "por defecto",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "por defecto",
                    "className": "list-group-item",
                  },
                  "ref": null,
                  "rendered": "por defecto",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "por defecto",
                    "className": "list-group-item",
                  },
                  "ref": null,
                  "rendered": "por defecto",
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "onChange": [Function],
                "rows": "5",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Enviar Solucion",
                "className": "btn btn-default",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Enviar Solucion",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-default"
                  onClick={[Function]}
                  type="button"
                >
                  Ver Soluciones propuestas
                </button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ver Soluciones propuestas",
                  "className": "btn btn-default",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Ver Soluciones propuestas",
                "type": "button",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
